### ISSUE TITLE
[FEATURE REQUEST] Estimate SMPL outputs

### ISSUE THREAD

**Issue by MelihDarcanxyz** [freemocap/freemocap#705]
**What is the use case for this feature?**
SMPL (SMPL+H, SMPL-X, etc.) model is the defacto body model for scientific virtual humans literature, at least for now. It would be highly beneficial for researchers if you could add features like SMPL estimation, utilizing Smplify, Smplify-X, etc.

**Additional context**
https://smpl.is.tue.mpg.de/
https://smplify.is.tue.mpg.de/

https://smpl-x.is.tue.mpg.de/

Perhaps how [AMASS](https://amass.is.tue.mpg.de/) handles SMPL estimation from MoCap markers would be beneficial.

**Comment by philipqueen** [comment]
Hi @MelihDarcanxyz, thanks for the issue.

Would you be willing to say more about how this will be beneficial to your research? Is it mostly for comparison to other research? 

We are working to provide other pose estimation options for V2 of Freemocap. We understand the limitations of MediaPipe and are interested in providing more state of the art options. Two alternative trackers, YOLO and OpenPose, are currently available to run as a scripts from `experimental/alternative_trackers` in the Freemocap directory. The only thing missing from these trackers is Blender support, all of the standard triangulation and post processing from Freemocap is available. 

Looking at SMPL and SMPLX, there are two hurdles that make them less attractive to implement. The first is the non-permissive license. We prefer permissive licenses that allow our users to do whatever they want with the data they collect. We have made exceptions for this in the past for academic standards like OpenPose though. The other hurdle is the requirement to register in order to be able to download the models. This makes it harder for us to provide the model, including likely support time trying to explain why the SMPL model isn't directly available like the rest, and conflicts with the free/open spirit of this project.

With that being said, it looks possible to implement SMPLX with Freemocap. But even looking through the examples comes with warnings of prosecution. So we would need a very good reason to implement SMPL instead of another state of the art pose estimator. Like you said, it seems possible to use the MOSH++ technique from AMASS to take our 3d data output and fit it to a SMPL model if that model type is required.

**Comment by MelihDarcanxyz** [comment]
Hi @philipqueen, thanks for detailed answer. Sorry for not being as detailed as possible above.

> Would you be willing to say more about how this will be beneficial to your research? Is it mostly for comparison to other research?

Sure, SMPL is utilized in SOTA human motion capture / generation research. This format is especially important because our joints / bones are not enough to represent us. We humans have different shapes. We also do not interact with the world around us with our bones, but with our skin. I mostly talked about the importance of SMPL until now, so sorry if this wasn't what you asked. If we talk about the practical implications, yes, comparison to other research is an important factor. Also, being able to utilize SMPL datasets like BEDLAM, AMASS, 3DPW, etc., for pretraining, then using custom SMPL dataset (created with freemocap) for fine-tuning would be beneficial.

Here, xrmocap implemented this before, however the project looks abandoned.

https://github.com/openxrlab/xrmocap

> The first is the non-permissive license.

I can totally see where you're coming from, and I understand that you want the project to be open as possible. I agree with your points. The only solution I can come up is to let the users know that it has a commercially restrictive license, and they should download the body model from the `<SMPL_WEBPAGE>` and put it into the `<SMPL_MODEL_PATH>`. But this may not be an ideal solution, this may not be a solution at all. It's up to you, after all.

I must say that I'm willing to help you if you ever decide to implement this feature, but I presume we will need to stay in contact because I may have lots of questions about freemocap.

**Comment by philipqueen** [comment]
> I don't think I understand this part. By SMPL do you refer to [smplify](https://github.com/openxrlab/xrmocap/blob/main/xrmocap/model/registrant/smplify.py) or [smplifyx](https://github.com/openxrlab/xrmocap/blob/main/xrmocap/model/registrant/smplifyx.py)? SMPL itself is the body model. It can be fit from point cloud, RGB monocular / multi-view images / videos. I think what you're referring to might be smplify. If I missed something, please let me know.
> 

I guess I was referring to smplify/smplify x. I was conflating those with SMPL, my bad.

> Is this required for multi person mocap?
>

Here I meant to combine multiple smplify views (from single camera views) into one multiview estimate. But to be clear, I think it's better to fit smpl to the 3d data, I was just trying to figure out what you meant. 

> There is a template SMPL mesh, `T` (which we download). Then we change its pose parameters either with rotation matrices, quaternions, or axis-angles. Then, `T` changes to specified pose. There are also beta parameters, that define the shape of the `T`. It's used to make the model taller, fatter, and whole lots of other features. So, the pose and beta parameters define the body model.
> 

Thanks, that helps clarify a bit


> Sure, I'll do my best. But again, just to speak the same language, I may ask lots of questions.

Given all of this, I do think finding a way to fit the current freemocap data to a SMPL model would be the best starting point. I don't have the time to do this myself, but I'm happy to assist you and answer any questions.

**Comment by philipqueen** [comment]
@MelihDarcanxyz great paper, there's a lot in there that's relevant to us (shame there is no associated repo, but looks like it would have the SMPL license anyways).

And I would be happy to help you make that integration. We are in the middle of a big rewrite that will hopefully make these integrations easier. Whenever you feel your work is ready for an integration, we can schedule a call to talk through it

**Comment by roaldarbol** [comment]
Just wanted to also highlight a different path that might be interesting from the OpenCapBench paper: https://arxiv.org/pdf/2406.09788v1.

They generate a SMPL mesh, and then detect the anatomical markers used in normal motion capture from the predicted mesh, which enables the use OpenSim for inverse kinematics etc. downstream. Could be an interesting path to pursue too!
